lib/puppet/provider/group/gpasswd.rb

```diff
@@ -0,0 +1,75 @@
+require 'puppet/provider/group/groupadd'
+
+Puppet::Type.type(:group).provide :gpasswd, :parent => Puppet::Type::Group::ProviderGroupadd do
+  require 'set'
+  require 'shellwords'
+
+  desc <<-EOM
+    Group management via `gpasswd`. This allows for local group
+    management when the users exist in a remote system.
+  EOM
+
+  commands  :addmember => 'gpasswd',
+            :delmember => 'gpasswd'
+
+  has_feature :manages_members unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
+
+  def addcmd
+    cmd = []
+    cmd << super.shelljoin
+
+    if @resource[:members]
+      cmd << @resource[:members].map{ |x|
+        [ command(:addmember),'-a',x,@resource[:name] ].shelljoin
+      }
+    end
+
+    # NOTE: This will create the group but may not add some members to
+    # your group if they do not exist on the system. Any member that
+    # can be added will be added in an attempt to do "the right
+    # thing".
+    cmd.join(' && ')
+  end
+
```

Comment:
The group type will call out to the provider (if it implements `members_insync?`) to check if the members are insync or not. You may need to implement it so that the resource is idempotent, something like:

```
  def members_insync?(current, should)
    return false unless current

    if @resource[:auth_membership]
      current == should
    else
      (current & should) == should
    end
```